packages/lesswrong/components/comments/CommentsNode.tsx

```diff
@@ -174,45 +191,49 @@ const CommentsNode = ({
 
   const scrollIntoView = useCallback((behavior: "auto"|"smooth"="smooth") => {
     if (!isInViewport()) {
-      const commentTop = getLandmarkY(commentIdToLandmark(comment._id));
-      if (commentTop) {
-        // Add window.scrollY because window.scrollTo takes a relative scroll distance
-        // rather than an absolute scroll position, and a +1 because of rounding issues
-        // that otherwise cause us to wind up just above the comment such that the ToC
-        // highlights the wrong one.
-        const y = commentTop + window.scrollY - getCurrentSectionMark() + 1;
-        window.scrollTo({ top: y, behavior });
-      }
+      scrollFocusOnElement({ id: comment._id, options: { behavior } });
     }
     setHighlighted(true);
     setTimeout(() => { //setTimeout make sure we execute this after the element has properly rendered
       setHighlighted(false);
     }, HIGHLIGHT_DURATION*1000);
   }, [comment._id]);
 
-  const handleExpand = async (event?: React.MouseEvent) => {
-    event?.stopPropagation()
+  const handleExpand = ({
+    event,
+    scroll = false,
+    scrollBehaviour = "smooth",
+  }: {
+    event?: React.MouseEvent;
+    scroll?: boolean;
+    scrollBehaviour?: "auto" | "smooth";
+  }) => {
+    event?.stopPropagation();
     if (isTruncated || isSingleLine) {
-      captureEvent('commentExpanded', {postId: comment.postId, commentId: comment._id})
+      captureEvent("commentExpanded", { postId: comment.postId, commentId: comment._id });
       setTruncated(false);
       setSingleLine(false);
       setTruncatedStateSet(true);
-      if (scrollOnExpand) {
-        scrollIntoView("auto") // should scroll instantly
+      if (scroll) {
+        scrollIntoView(scrollBehaviour);
       }
     }
-  }
+  };
+
+  // If not using in-context comments, scroll to top when the `commentId` query changes
+  useEffect(() => {
+    if (commentPermalinkStyleSetting.get() === 'top' && !noAutoScroll && comment && linkedCommentId === comment._id) {
+      window.scrollTo({top: 0})
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [linkedCommentId]);
```

Comment:
I guess it actually is for LW, because the `ScrollToTop` callback still runs on changes in the query. But in general I think it makes sense to have this because you could have a forum that has `commentPermalinkStyleSetting` set to 'top' and doesn't `ScrollToTop` on query changes